readme.md

```diff
@@ -394,71 +397,121 @@ resolved in Node 8.3 with V8 6.0.
 
 _Edit: Jakob Kummerow noted in [http://disq.us/p/1kvomfk](http://disq.us/p/1kvomfk) that Turbofan
 can optimize away the object allocation in this specific microbenchmark,
-leading to incorrect results.
-The creating object benchmark has been split into creating objects and
-creating objects that does not "escape"._
+leading to incorrect results so this article has been edited accordingly._
+
+#### Object allocation elimination
+
+While collating our results for this article, we came across a really
+neat optimization that Turbofan brings to a certain category of
+object allocation. Originally we mistook this for all object allocation, but
+thanks to input from the V8 team we've come to understand the circumstances
+which this optimization relates to.
+
+In the previous **Object allocation** benchmarks we assign a variable, set it
+to `null` and then reassign the variable a bunch of times to avoid triggering
+the special optimization case which we're going to look at now.
+
+In this benchmark we look at three cases:
+
+* allocating objects using object literals (*literal*)
+* allocating objects from an EcmaScript 2015 Class (*class*)
+* allocating objects from a constructor function (*constructor*)
+
+However the difference is, the reference to the object is not overwritten
+with additional object allocations, and the object is passed to another function
+which performs a task on the object.
+
+Let's take a look at the results!
+
+**Code:** <https://github.com/davidmarkclements/v8-perf/blob/master/bench/object-creation-inlining.js>
+
+![](graphs/hardcoded-object-creation-bar.png)
+
+Notice how V8 6.0 (Node 8.3) and 6.1 (Node 9) yield a massive jump in speed for this case,
+over 500 million operations per second, mainly because nothing is really happening once
+Turbofan applies the optimization. In this particular scenario, Turbofan is able to
+*abstract away* the object allocation allowing subsequent execution of logic that processes the object
+to essentially becomes a no-op.
+
+The benchmark code still doesn't fully demonstrate what it takes to trigger this
+optimization (because it's more about what's not there than what is) but the conditions it
+takes for this optimization to be applied by Turbofan are as follows.
 
-#### Creating objects that does not "escape"
+First, do not alter the content of the variable. That is don't reassign a variable with a new object
+once that variable has already been assigned.
 
-An extremely common pattern in JS is to create an "option object" to
-specify some options to a function call, e.g.:
+Secondly, the object must not outlive the function it was created in. That means, there should be
+no reference to that object after every function from that point in the stack has completed. The object
+*can* be passed to other functions, but if we add that object to a `this` context, or assign it
+to an outer scoped variable, or use it as the return value from the current function, or add it to
```

Comment:
This is in accurate. Counter-example:
```
function create() {
  return {o: 1}
}
function drop() {
  var a = create();
}
```
The object allocation will be optimized away. That's possible because the compiler decided to inline `create` into `drop`.